docs/doxygen/porting.txt

```diff
@@ -0,0 +1,77 @@
+/**
+@page mqtt_porting Porting Guide
+@brief Guide for porting MQTT to a new platform.
+
+A port to a new system must provide the following components:
+1. [Configuration Macros](@ref porting_config)
+2. [Transport Interface](@ref porting_transport)
+3. [Time Function](@ref porting_time)
+
+@section porting_config Configuration Macros
+@brief Settings that must be set as macros in the config header `core_mqtt_config.h`, or passed in as compiler options
+
+@see [Configurations](@ref core_mqtt_config)
+
+@note Regardless of whether the following macros are defined in `core_mqtt_config.h` or passed as compiler options,
+the `core_mqtt_config.h` file must be included to build the MQTT library.
```

Comment:
Similarly for this line. The custom config is not strictly required anymore

Reply:
Removed `@note`